packages/web/components/bridge/amount-screen.tsx

```diff
@@ -0,0 +1,425 @@
+import { Menu, MenuButton, MenuItem, MenuItems } from "@headlessui/react";
+import { CoinPretty, Dec, DecUtils, PricePretty } from "@keplr-wallet/unit";
+import { DEFAULT_VS_CURRENCY } from "@osmosis-labs/server";
+import { isNumeric } from "@osmosis-labs/utils";
+import classNames from "classnames";
+import Image from "next/image";
+import { FunctionComponent, ReactNode, useState } from "react";
+
+import { Icon } from "~/components/assets";
+import { BridgeNetworkSelect } from "~/components/bridge/bridge-network-select";
+import { MoreBridgeOptions } from "~/components/bridge/more-bridge-options";
+import { InputBox } from "~/components/input";
+import { Spinner } from "~/components/loaders";
+import { Tooltip } from "~/components/tooltip";
+import { Button } from "~/components/ui/button";
+import { useTranslation } from "~/hooks";
+import { useStore } from "~/stores";
+import { trimPlaceholderZeros } from "~/utils/number";
+import { api } from "~/utils/trpc";
+
+interface AmountScreenProps {
+  type: "deposit" | "withdraw";
+}
+
+export const AmountScreen = ({ type }: AmountScreenProps) => {
+  const {
+    accountStore,
+    chainStore: {
+      osmosis: { chainId },
+    },
+  } = useStore();
+  const wallet = accountStore.getWallet(chainId);
+  const [isMoreOptionsVisible, setIsMoreOptionsVisible] = useState(false);
+  const { t } = useTranslation();
+
+  const { data: asset, isLoading } = api.edge.assets.getAssetWithPrice.useQuery(
+    {
+      findMinDenomOrSymbol: "USDC",
+    }
+  );
```

Comment:
Remember, we're going to eventually have to fetch all available source tokens for _all_ variants on Osmosis. i.e. all USDCs on Osmosis (Noble, Axelar, etc.). Then, we will reduce those into chain and asset options that a user can select from. I'm posting this here to keep in mind and make sure we're on same page.

Reply:
Yes! I'm on the same page. I'm just hard-coding this to complete the layout. I expect to refine all of this once we are actually implementing quotes and logic.